File: frontend/src/types/project-github.ts
export interface ProjectReadme {
  content: string
  source: "cache" | "github"
  path?: string | null
  is_default?: boolean
  cached_at?: string | null
  github_sha?: string | null
  content_changed?: boolean
}

export interface ProjectReleaseAsset {
  name: string
  size: number | null
  download_count: number
  browser_download_url: string
  updated_at: string | null
}

export interface ProjectRelease {
  tag_name: string
  name: string | null
  body: string | null
  published_at: string | null
  html_url: string | null
  prerelease: boolean
  draft: boolean
  assets?: ProjectReleaseAsset[]
}

export interface ProjectReleasesResponse {
  items: ProjectRelease[]
  source?: "cache" | "github"
  cached_at?: string | null
  content_changed?: boolean
}

export type ProjectDetailTab = "readme" | "release" | "notes"

export const PROJECT_DETAIL_TABS: ProjectDetailTab[] = ["readme", "release", "notes"]

export function parseProjectDetailTab(value: string | null): ProjectDetailTab {
  if (value === "release" || value === "notes") {
    return value
  }
  return "readme"
}

File: backend/app/schemas/project_github.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ProjectReadmeRead(BaseModel):
    content: str
    source: Literal["cache", "github"] = "github"
    path: str | None = None
    is_default: bool = True
    cached_at: datetime | None = None
    github_sha: str | None = None
    content_changed: bool = False


class ProjectReleaseAssetRead(BaseModel):
    name: str
    size: int | None = None
    download_count: int = 0
    browser_download_url: str
    updated_at: datetime | None = None


class ProjectReleaseRead(BaseModel):
    tag_name: str
    name: str | None = None
    body: str | None = None
    published_at: datetime | None = None
    html_url: str | None = None
    prerelease: bool = False
    draft: bool = False
    assets: list[ProjectReleaseAssetRead] = Field(default_factory=list)


class ProjectReleasesRead(BaseModel):
    items: list[ProjectReleaseRead] = Field(default_factory=list)
    source: Literal["cache", "github"] = "github"
    cached_at: datetime | None = None
    content_changed: bool = False


class GithubRepoPreviewRead(BaseModel):
    """添加项目前从 GitHub 拉取的仓库简介预览。"""

    full_name: str | None = None
    description: str | None = None
    fetched: bool = False
    error: str | None = Field(
        None,
        description="无法获取时的用户可读说明（网络、Token、仓库不存在等）",
    )
